Description: Remove verbose messages printed before reading configuration
 In some ways this is awkward because it makes debugging harder, but it's a
 requirement for a smooth-looking boot process; we may be able to do better in
 future.  Upstream doesn't want this, though.
 .
 We disable the cursor as well, for similar reasons of tidiness.
Author: Colin Watson <cjwatson@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/386922
Bug-Ubuntu: https://bugs.launchpad.net/bugs/861048
Forwarded: not-needed
Last-Update: 2012-09-06

Index: b/grub-core/kern/main.c
===================================================================
--- a/grub-core/kern/main.c
+++ b/grub-core/kern/main.c
@@ -207,13 +207,14 @@
 void __attribute__ ((noreturn))
 grub_main (void)
 {
+  struct grub_term_output *term;
+
   /* First of all, initialize the machine.  */
   grub_machine_init ();
 
-  /* Hello.  */
-  grub_setcolorstate (GRUB_TERM_COLOR_HIGHLIGHT);
-  grub_printf ("Welcome to GRUB!\n\n");
-  grub_setcolorstate (GRUB_TERM_COLOR_STANDARD);
+  /* Disable the cursor until we need it.  */
+  FOR_ACTIVE_TERM_OUTPUTS(term)
+    grub_term_setcursor (term, 0);
 
   /* Load pre-loaded modules and free the space.  */
   grub_register_exported_symbols ();
@@ -232,5 +233,10 @@
 
   grub_load_config ();
   grub_load_normal_mode ();
+
+  /* If we have to enter rescue mode, enable the cursor again.  */
+  FOR_ACTIVE_TERM_OUTPUTS(term)
+    grub_term_setcursor (term, 1);
+
   grub_rescue_run ();
 }
Index: b/grub-core/kern/rescue_reader.c
===================================================================
--- a/grub-core/kern/rescue_reader.c
+++ b/grub-core/kern/rescue_reader.c
@@ -77,8 +77,6 @@
 void __attribute__ ((noreturn))
 grub_rescue_run (void)
 {
-  grub_printf ("Entering rescue mode...\n");
-
   while (1)
     {
       char *line;
Index: b/grub-core/normal/menu.c
===================================================================
--- a/grub-core/normal/menu.c
+++ b/grub-core/normal/menu.c
@@ -669,12 +669,9 @@
 
 /* Callback invoked immediately before a menu entry is executed.  */
 static void
-notify_booting (grub_menu_entry_t entry,
+notify_booting (grub_menu_entry_t entry __attribute__((unused)),
 		void *userdata __attribute__((unused)))
 {
-  grub_printf ("  ");
-  grub_printf_ (N_("Booting `%s'"), entry->title);
-  grub_printf ("\n\n");
 }
 
 /* Callback invoked when a default menu entry executed because of a timeout
